tests/e2e/common/kubernetes.go

```diff
@@ -164,27 +165,53 @@ func CleanupNode(ctx context.Context, cs clientset.Interface) {
 	}
 }
 
-func CleanupNodeFeatureRules(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
-	// Drop NodeFeatureRule objects
+// cleanupNodeFeatures deletes all NodeFeature objects in the given namespace
+func cleanupNodeFeatures(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
+	nfs, err := cli.NfdV1alpha1().NodeFeatures(namespace).List(ctx, metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		// Omitted error, nothing to do.
+		return
+	}
+	Expect(err).NotTo(HaveOccurred())
+
+	if len(nfs.Items) != 0 {
+		By("Deleting NodeFeature objects from namespace " + namespace)
+		for _, nf := range nfs.Items {
+			err = cli.NfdV1alpha1().NodeFeatures(namespace).Delete(ctx, nf.Name, metav1.DeleteOptions{})
+			if errors.IsNotFound(err) {
+				// Omitted error
+				continue
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+			}
+		}
+	}
+}
+
+// cleanupNodeFeatureRules deletes all NodeFeatureRule objects
+func cleanupNodeFeatureRules(ctx context.Context, cli *nfdclient.Clientset) {
 	nfrs, err := cli.NfdV1alpha1().NodeFeatureRules().List(ctx, metav1.ListOptions{})
+	if errors.IsNotFound(err) {
+		// Omitted error, nothing to do.
+		return
+	}
 	Expect(err).NotTo(HaveOccurred())
 
 	if len(nfrs.Items) != 0 {
 		By("Deleting NodeFeatureRule objects from the cluster")
 		for _, nfr := range nfrs.Items {
 			err = cli.NfdV1alpha1().NodeFeatureRules().Delete(ctx, nfr.Name, metav1.DeleteOptions{})
-			Expect(err).NotTo(HaveOccurred())
+			if errors.IsNotFound(err) {
+				// Omitted error
+				continue
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+			}
 		}
 	}
+}
 
-	nfs, err := cli.NfdV1alpha1().NodeFeatures(namespace).List(ctx, metav1.ListOptions{})
-	Expect(err).NotTo(HaveOccurred())
-
-	if len(nfs.Items) != 0 {
-		By("Deleting NodeFeature objects from namespace " + namespace)
-		for _, nf := range nfs.Items {
-			err = cli.NfdV1alpha1().NodeFeatures(namespace).Delete(ctx, nf.Name, metav1.DeleteOptions{})
-			Expect(err).NotTo(HaveOccurred())
-		}
-	}
+func CleanupNFDObjects(ctx context.Context, cli *nfdclient.Clientset, namespace string) {
```

Comment:
nit: If we have the public function before the others the diff might align better and the file would be easier to skim from top to bottom.